.github/workflows/ci.yml

```
@@ -0,0 +1,108 @@
+# This workflow will build a Java project with Maven
+# For more information see: https://help.github.com/actions/language-and-framework-guides/building-and-testing-java-with-maven
+
+name: XNIO CI
+
+on:
+  pull_request:
+    types: [opened, synchronize, reopened, ready_for_review]
+
+jobs:
+  build-all:
+    name: Compile (no tests) with JDK 8
+    runs-on: ubuntu-latest
+    steps:
+    - uses: n1hility/cancel-previous-runs@v2
+      with:
+          token: ${{ secrets.GITHUB_TOKEN }}
+    - uses: actions/cache@v1
+      with:
+        path: ~/.m2/repository
+        key: m2-${{ hashFiles('**/pom.xml') }}
+        restore-keys: |
+            m2-
+    - uses: actions/checkout@v2
+    - name: Host information
+      run: netstat -anp
+      shell: bash
+    - name: Set up JDK 8
+      uses: joschi/setup-jdk@v2
+      with:
+        java-version: 8
```

```suggestion
   - uses: AdoptOpenJDK/install-jdk@v1
        name: set up JDK 8
        with:
          version: 8
```

The [AdoptOpenJDK one](https://github.com/AdoptOpenJDK/install-jdk) is compatible with the jboss/smallrye-parent method of MR JAR building (it properly sets the necessary env vars).  XNIO isn't an MR JAR yet but the time will come, I expect.

---

Thanks, I hadn't seen the AdoptOpenJDK one, will update to use that.